pkg/helpers/helpers.go

```diff
@@ -736,9 +736,9 @@ func UpdateKlusterletRelatedResourcesFn(relatedResources ...operatorapiv1.Relate
 }
 
 // KlusterletNamespace returns the klusterletNamespace to deploy the agents.
-// Note in Detached mode, the specNamespace will be ignored.
+// Note in Hosted mode, the specNamespace will be ignored.
 func KlusterletNamespace(klusterlet *operatorapiv1.Klusterlet) string {
-	if klusterlet.Spec.DeployOption.Mode == operatorapiv1.InstallModeDetached {
+	if klusterlet.Spec.DeployOption.Mode == operatorapiv1.InstallModeHosted {
```

Comment:
hrm, this would still need consider detached?

Reply:
I'm not sure about the "read from klusterlet resource" case, will this case have affect on this function tool?

Reply:
Updated  with Detached also considered.